doc/howtos/software-update-server.rst

```diff
@@ -0,0 +1,57 @@
+.. _software-update-server:
+
+Software Update: Building an Ostro |trade| OS  Repository
+#########################################################
+
+This technical note provides instructions supporting the creation
+of a repository for feeding updates to devices based on Ostro OS.
+
+Prerequisites
+=============
+
+- additional (fast) disk space: the SW Update tool is provided as
+  additional layer, therefore it doesn't require any additional
+  element.
+  However, the process of generating the bundles/updates can be
+  punishing for the disk.
+  Using a fast disk (ex: SSD/RAID) can bring dramatic improvements.
+  An execution round will also produce various rootfs directory as
+  intermediate artefacts.
+  The actual disk footprint depends on the bundles selected/defined,
+  but it will definitely be larger, compared to the generation of a
+  simple build, like one might be used to, with a typical distribution
+  based on OECore/YP
+- signing keys (alternatively testing keys from the :file:`swupd-server`
+  project can be used).
+
+Bundles definition
+==================
+
+The extensive documentation for using SW Update in YP context can be
+found in the `swupd layer`_ and will not be replicated here.
+These instructions are based on the assumption that the work will happen
+on top of an image flavor that has swupd enabled.
+
+These are the main steps:
+
+- set the `OS_VERSION` variable to an integer value matching `VERSION_ID`
+  in the os-release recipe.
+  This number must be increased before each build that should generate
+  swupd update artefacts.
+- assign a list of bundle names to `SWUPD_BUNDLES` i.e:
+
+    ```SWUPD_BUNDLES = "feature_one feature_two"```
+
+- for each named bundle, setup a varflag of `BUNDLE_CONTENTS` which
+  matches the bundle name and initialize it with a list of packages whose
+  content must be included in the bundle.
+  Example:
+    ```BUNDLE_CONTENTS[feature_one] = "package_one package_three package_six"```
+
+
+Creating a Software Update Repo
+===============================
+
+[Pending: will come in a separate PR]
+
+.. _`swupd layer`: http://git.yoctoproject.org/cgit/cgit.cgi/meta-swupd/tree/docs/GUIDE.md
```

Comment:
Not sure when but the file name was changed from GUIDE.md to Guide.md so we need to update this link to http://git.yoctoproject.org/cgit/cgit.cgi/meta-swupd/tree/docs/Guide.md